### PR TITLE
[TIMOB-24589] TiAPI: Remove Ti.Yahoo

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24589

> `no tests` since there were no tests and we are..well...removing it 😙

### CODE REVIEW

File: apidoc/Titanium/Yahoo/Yahoo.yml [hunk 1]
```diff
@@ -14,6 +14,10 @@ methods:
       - name: callback
         summary: the function to execute when the query completes.
         type: Callback<YQLResponse>
+deprecated:
+    since: "8.0.0"
```
Comment (sgtcoolguy) [inline apidoc/Titanium/Yahoo/Yahoo.yml]: didn't we deprecate this before?

**Review by sgtcoolguy — APPROVED** [review]
Should have been killed off a very long time ago...

File: apidoc/Titanium/Yahoo/Yahoo.yml (hunk 1 above)
Comment (hansemannn) [reply apidoc/Titanium/Yahoo/Yahoo.yml]: I think we planned to in 7 but never came to it

File: android/modules/contacts/src/java/ti/modules/titanium/contacts/CommonContactsApi.java [hunk 2]
```diff
@@ -143,8 +143,6 @@ protected static String getImTextType(int type)
 			key = "custom";
 		} else if (type == ContactsContract.CommonDataKinds.Im.PROTOCOL_MSN) {
 			key = "msn";
-		} else if (type == ContactsContract.CommonDataKinds.Im.PROTOCOL_YAHOO) {
```
Comment (sgtcoolguy) [inline android/modules/contacts/src/java/ti/modules/titanium/contacts/CommonContactsApi.java]: Should this be removed? Does Yahoo still have IM? I don't think it's related to our Ti.Yahoo API.

File: android/modules/contacts/src/java/ti/modules/titanium/contacts/ContactsApiLevel5.java [hunk 3]
```diff
@@ -286,8 +286,6 @@ protected int processIMProtocol(String serviceName)
 			return Im.PROTOCOL_QQ;
 		} else if (serviceName.equals("Skype")) {
 			return Im.PROTOCOL_SKYPE;
-		} else if (serviceName.equals("Yahoo")) {
```
Comment (sgtcoolguy) [inline android/modules/contacts/src/java/ti/modules/titanium/contacts/ContactsApiLevel5.java]: Should this be removed? Does Yahoo still have IM? I don't think it's related to our Ti.Yahoo API.

**Review by sgtcoolguy — CHANGES_REQUESTED** [review]
I think you removed a couple yahoo related things from the contacts API that wren't tied to our Ti.Yahoo API in any way and should have been retained. 

Also, This should probably be targeting next branch now.

**Review by sgtcoolguy — APPROVED** [review]

**Pull request merged**